src/psyclone/tests/psyir/nodes/array_member_test.py

```diff
@@ -167,7 +167,7 @@ def test_am_matching_access():
     ref1 = nodes.StructureReference.create(sym1, [("data", [one.copy()]),
                                                   ("xobs", [one.copy()])])
     ref2 = nodes.StructureReference.create(sym1, [("data", [one.copy()])])
-    assert ref2.member._matching_access(ref1) is True
+    assert ref1.member._matching_access(ref2) is True
```

Comment:
Why was the previous one (with swapped refs) working? Should we check that `ref2.member._matching_access(ref1)` is False now (or it isn't?)

Reply:
I'm not sure why it was (or rather, wasn't) working before. However, as you suggest I've added a check that `ref2.member.is_same_array(ref1)` is False and it is.